database_manager/src/run_server.cpp

```diff
@@ -1775,11 +1771,7 @@ void update_world_states(ros::NodeHandle& node, std::vector<ToasterFactReader*>
         for (int i = 0; i < newState.size(); ++i) {
             if ((previousFactsState[j].subjectId == newState[i].subjectId)
                     && (previousFactsState[j].property == newState[i].property)
-                    && (previousFactsState[j].propertyType == newState[i].propertyType)
-                    && (previousFactsState[j].subProperty == newState[i].subProperty)
-                    && (previousFactsState[j].subjectOwnerId == newState[i].subjectOwnerId)
-                    && (previousFactsState[j].targetId == newState[i].targetId)
-                    && (previousFactsState[j].targetOwnerId == newState[i].targetOwnerId)) {
+                    && (previousFactsState[j].targetId == newState[i].targetId)) {
```

Comment:
I don't really get why you removed those lines?
There could be 2 identical facts except for there subproperty.

As an example, isMovingToward is computed in 2 different ways, and the difference is in subProperty.
So They could be mistaken one for another if you remove those lines...